### DISCUSSION TITLE
Supporting `aarch64-apple-darwin` on the Nix flake packages

### DISCUSSION THREAD

**Discussion by oxcabe** [ghostty-org/ghostty#3800] [Feature Requests, Ideas] [answered]
Trying to install the Nix Flake packages from this repo on a Nix profile in MacOS will raise this error:

```text
error: Package ‘ghostty-1.0.1’ in /nix/store/jcwklmiadwm351xdvgbvmb1wjgh4zybg-source/nix/package.nix:113 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "x86_64-linux"
           "aarch64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.
```

After having a quick look into the repo, it seems the `aarch64-apple-darwin` is not supported:

https://github.com/ghostty-org/ghostty/blob/ca7c9547129cb156417f7aa50cb390edecb685d7/nix/package.nix#L190-L195

I'm not sure if this is deliberate or not, and just wanted to ask before moving this into a potentially unnecessary issue.

Are MacOS releases only expected to be downloaded as signed binaries from the releases? If not, could `aarch64-apple-darwin` make it into the supported platforms list?

**Comment by jcollie** [comment, marked as answer]
Building a macOS binary requires running `xcodebuild` which doesn't work inside a Nix development environment. There's some work going on in the NixOS community to get a Darwin build within Nix working but there are some features that Ghostty takes advantage of that require a signed application. So it's likely that the best method for macOS users to install Ghostty will be to download the signed binaries.

**Reply by oxcabe** [reply]
Oh, I see, thanks for the prompt reply and explanation!

**Reply by Eveeifyeve** [reply]
There is already a duplicate discussion here: https://github.com/ghostty-org/ghostty/discussions/2824

**Reply by majordoobie** [reply]
Thanks for the reply!
